src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.Created.cs

```diff
@@ -111,4 +111,73 @@ public static void FileSystemWatcher_Created_Negative()
             Utility.ExpectNoEvent(eventOccured, "created");
         }
     }
+
+    [Fact]
+    public static void FileSystemWatcher_Created_NestedDirectories()
+    {
+        using (var dir = Utility.CreateTestDirectory())
+        using (var watcher = new FileSystemWatcher())
+        {
+            AutoResetEvent eventOccured = Utility.WatchForEvents(watcher, WatcherChangeTypes.Created);
+
+            watcher.Path = Path.GetFullPath(dir.Path);
+            watcher.Filter = "*";
+            watcher.IncludeSubdirectories = true;
+            watcher.EnableRaisingEvents = true;
+
+            using (var firstDir = new TemporaryTestDirectory(Path.Combine(dir.Path, "dir1")))
```

Comment:
There's a lot of repetition in the structure of these tests.  If you're able to condense them due to my previous suggestion, great.  If such repetition still remains, consider turning this into a helper function, e.g. pseudo-code:
```C#
private static void TestNestedDirectories(WatcherChangeTypes change, Action<AutoResetEvent, TemporaryTestDirectory> action)
{
    using (var dir = Utility.CreateTestDirectory())
    using (var watcher = new FileSystemWatcher())
    using (AutoResetEvent eventOccurred = Utility.WatchForEvents(watcher, change))
    {
        watcher.Path = ...;
        ... // watcher setup

        using (var firstDir = new TemporaryTestDirectory(Path.Combine(dir.Path, "dir1")))
        {
            WaitForEventForFirstDir(); // whatever this ends up being
            using (var secondDir = new TemporaryTestDirectory(Path.Combine(firstDir.Path, "dir2")))
            {
                action(eventOccurred, secondDir);
            }
        }
    }
}
```
or something like that.